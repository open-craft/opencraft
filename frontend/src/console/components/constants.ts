export enum RedeploymentStatus {
  REDEPLOYING = "REDEPLOYING",
  UPDATED = "UPDATED",
  FAILED = "FAILED",
  MAINTENANCE = "MAINTENANCE",
  NO_STATUS = "NO_STATUS"
}
