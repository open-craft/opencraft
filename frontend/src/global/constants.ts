export const OCIM_API_BASE =
  process.env.REACT_APP_OCIM_API_BASE || 'http://localhost:5000';

export const CONTACT_US_LINK = process.env.REACT_APP_CONTACT_US_LINK || '/#';
export const ENTERPRISE_COMPARISON_LINK =
  process.env.REACT_APP_ENTERPRISE_COMPARISON_LINK || '/#';
export const TOS_LINK = process.env.REACT_APP_TOS_LINK || '/#';
export const PRIVACY_POLICY_LINK =
  process.env.REACT_APP_PRIVACY_POLICY_LINK || '/#';

export const INTERNAL_DOMAIN_NAME =
  process.env.REACT_APP_INTERNAL_DOMAIN_NAME || '.opencraft.hosting';

export const GANDI_REFERRAL_LINK =
  process.env.REACT_APP_INTERNAL_DOMAIN_NAME || 'https://gandi.link/f/ba351d73';

export interface StringIndexedArray {
  [key: string]: any;
}

export enum RegistrationSteps {
  FIRST_STEP = 0,
  DOMAIN = 0,
  CUSTOM_DOMAIN = 1,
  INSTANCE = 2,
  ACCOUNT = 3,
  CONGRATS = 4,
  LAST_STEP = 4
}

export const ROUTES = {
  Auth: {
    LOGIN: '/login',
    LOGOUT: '/logout',
    PASSWORD_FORGOTTEN: '/password-forgotten',
    PASSWORD_RESET: '/password-reset/:token'
  },
  Console: {
    HOME: '/console',
    CUSTOM_PAGES: '/console/custom-pages/:pageName',
    THEME_PREVIEW_AND_COLORS: '/console/theming/preview-and-colors',
    LOGOS: '/console/theming/logos',
    THEME_BUTTONS: '/console/theming/buttons',
    THEME_NAVIGATION: '/console/theming/navigation',
    THEME_FOOTER: '/console/theming/footer',
    INSTANCE_SETTINGS_GENERAL: '/console/settings/general',
    HERO: '/console/theming/hero'
  },
  Registration: {
    HOME: '/registration',
    DOMAIN: '/registration/domain',
    CUSTOM_DOMAIN: '/registration/custom-domain',
    INSTANCE: '/registration/instance',
    ACCOUNT: '/registration/account',
    CONGRATS: '/registration/congrats'
  },
  Error: {
    UNKNOWN_ERROR: '/error'
  }
};

export const REGISTRATION_STEPS = [
  ROUTES.Registration.DOMAIN,
  ROUTES.Registration.CUSTOM_DOMAIN,
  ROUTES.Registration.INSTANCE,
  ROUTES.Registration.ACCOUNT,
  ROUTES.Registration.CONGRATS
];

export const AVAILABLE_CUSTOM_PAGES = [
  'about',
  'contact',
  'donate',
  'tos',
  'honor',
  'privacy'
];

export const LMS_CUSTOM_PAGE_LINK_MAP: StringIndexedArray = {
  about: 'about',
  contact: 'support/contact_us',
  donate: 'donate',
  tos: 'tos',
  honor: 'honor',
  privacy: 'privacy'
};
