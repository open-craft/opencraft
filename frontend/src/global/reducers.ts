import { connectRouter } from 'connected-react-router';
import { History } from 'history';
import { combineReducers } from 'redux';
import { registrationReducer } from '../registration/reducers';
import { loginStateReducer } from '../auth/reducers';
import { consoleReducer } from '../console/reducers';
import { uiStateReducer } from '../ui/reducers';

export const createRootReducer = (history: History) =>
  combineReducers({
    console: consoleReducer,
    loginState: loginStateReducer,
    registration: registrationReducer,
    router: connectRouter(history),
    ui: uiStateReducer
  });
