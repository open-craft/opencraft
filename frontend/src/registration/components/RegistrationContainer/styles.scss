@import '~styles/theme';

.registration-container {
  @include add-background-image;
}
