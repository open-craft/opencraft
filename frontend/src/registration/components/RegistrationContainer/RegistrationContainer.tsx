import * as React from 'react';
import { Col } from 'react-bootstrap';
import { RegistrationRoutes } from 'routes/registration';
import './styles.scss';

export const RegistrationContainer: React.FC = () => (
  <Col className="registration-container">
    <RegistrationRoutes />
  </Col>
);
