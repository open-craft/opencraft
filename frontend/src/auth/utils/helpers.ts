import { V2Api } from 'global/api';

export const checkAuthAndRefreshToken = async () => {
  const refreshToken = localStorage.getItem('token_refresh');
  let response = null;

  if (refreshToken) {
    try {
      response = await V2Api.authRefreshCreate({
        data: { refresh: refreshToken }
      });
    } catch (e) {
      return false;
    }
  }

  if (response) {
    localStorage.setItem('token_access', response.access);
    return true;
  }

  return false;
};
