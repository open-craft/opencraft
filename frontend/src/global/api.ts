import { OCIM_API_BASE } from 'global/constants';
import { Configuration, V2Api as _V2Api } from 'ocim-client';
import { performLogout } from 'auth/actions';
import { checkAuthAndRefreshToken } from '../auth/utils/helpers';

const getApiKey = () => {
  const authToken = localStorage.getItem('token_access');
  // The current version of the API client requires us to specify the word "Bearer"
  // The previous version did not.
  return authToken ? `Bearer ${authToken}` : '';
};

const config = new Configuration({
  basePath: `${OCIM_API_BASE}/api`,
  apiKey: (name: string, scopes?: string[]) => getApiKey(),
  middleware: [
    {
      post: async context => {
        if (context.response.status === 403) {
          // Failed requests return 403 and mean that the access token is
          // expired, so we trigger a page refresh.
          const authenticated = await checkAuthAndRefreshToken();
          if (!authenticated) {
            window.location.reload(false);
          } else {
            // Retry request after updating API key
            const requestContext = context.init;
            requestContext.headers = {
              Authorization: getApiKey()
            };
            return context.fetch(context.url, requestContext);
          }
        } else if (context.response.status === 401) {
          // If the refresh endpoint fails, it means that the refresh
          // is expired, and the user is effectively logged out.
          performLogout();
        }
        return context.response;
      }
    }
  ]
});

export const V2Api = new _V2Api(config);

if (process.env.NODE_ENV === 'development') {
  // For development and debugging purposes, expose the API client to the browser console.
  (window as any).V2Api = V2Api;
}
