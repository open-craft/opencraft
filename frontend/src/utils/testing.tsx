import { createMemoryHistory } from 'history';
import * as React from 'react';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import * as renderer from 'react-test-renderer';
import { applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import { createRootReducer } from '../global/reducers';

export const setupComponentForTesting = (
  reactContent: JSX.Element,
  storeContents = {}
) => {
  const middleware = applyMiddleware(thunk);

  const store = createStore(
    createRootReducer(createMemoryHistory()),
    storeContents,
    middleware
  );

  return renderer.create(
    <IntlProvider textComponent={React.Fragment} locale="en">
      <Provider store={store}>
        <MemoryRouter>{reactContent}</MemoryRouter>
      </Provider>
    </IntlProvider>
  );
};
