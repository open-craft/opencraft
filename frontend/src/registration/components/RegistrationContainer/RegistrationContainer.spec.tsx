import { RegistrationSteps } from "global/constants";
import React from 'react';
import { setupComponentForTesting } from "utils/testing";
import { RegistrationContainer } from './RegistrationContainer';

it('renders without crashing', () => {
    const tree = setupComponentForTesting(
        <RegistrationContainer step={RegistrationSteps.THEME} submitRegistration={jest.fn()}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
