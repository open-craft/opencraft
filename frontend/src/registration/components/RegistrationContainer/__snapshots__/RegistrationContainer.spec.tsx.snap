// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="registration-container col"
>
  <h1>
    This is the registration page 
  </h1>
</div>
`;
